analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementISerializableCorrectly.cs

```diff
@@ -56,7 +57,9 @@ public class ImplementISerializableCorrectly : SonarDiagnosticAnalyzer
                     implementationErrors.AddRange(CheckGetObjectData(typeDeclarationSyntax, typeSymbol, getObjectData));
                     if (implementationErrors.Any())
                     {
-                        c.ReportIssue(Diagnostic.Create(Rule, typeDeclarationSyntax.Identifier.GetLocation(), implementationErrors.ToAdditionalLocations(), implementationErrors.ToProperties()));
+                        var details = implementationErrors.Aggregate(new StringBuilder(), (sb, error) => sb.Append($"{error.Message} "), sb => sb.ToString().TrimEnd());
+                        var rule = Diagnostic.Create(Rule, typeDeclarationSyntax.Identifier.GetLocation(), implementationErrors.ToAdditionalLocations(), implementationErrors.ToProperties(), details);
+                        c.ReportIssue(rule);
```

Comment:
I am not convinced that appending the secondary messages is a good idea. I think it brings confusion as the secondary messages are not at the right location.
For instance, I was confused by the following message raised at the class name level:
`Update this implementation of 'ISerializable' to conform to the recommended serialization pattern. Invoke 'base.GetObjectData(SerializationInfo, StreamingContext)' in this method.` (see [here](https://github.com/SonarSource/sonar-dotnet/blob/9951a3c01954f812e3038c2afe21f4c37a025adf/analyzers/its/expected/Net5/Net5--net5.0-S3925.json#L5))

I might be missing some context though.